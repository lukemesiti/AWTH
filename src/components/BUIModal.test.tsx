import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { BUIModal } from ".";
import { useState } from "react";
import userEvent from "@testing-library/user-event";

global.ResizeObserver = class FakeResizeObserver {
  observe() {}
  disconnect() {}
  unobserve() {}
};

const testModalId = "test-modal";
const openModalButton = "Open modal";

const TestModal: React.FC = () => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <>
      <button onClick={() => setIsOpen(true)}>{openModalButton}</button>
      <BUIModal
        title="Modal title"
        isOpen={isOpen}
        handleClose={() => setIsOpen(false)}
        testId={testModalId}
      >
        <div>modal content</div>
        <button onClick={() => setIsOpen(false)}>Close Modal</button>
      </BUIModal>
    </>
  );
};

describe("BUIModal", () => {
  it("should render with title and children components", async () => {
    // Arrange
    const user = userEvent.setup();
    render(<TestModal />);

    // Act
    await user.click(screen.getByText(openModalButton));

    // Assert
    expect(screen.getByTestId(testModalId)).toBeVisible();
    expect(screen.getByTestId(testModalId)).toMatchSnapshot();
  });
});
