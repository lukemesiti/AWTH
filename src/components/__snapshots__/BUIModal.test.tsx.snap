// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BUIModal > should render with title and children components 1`] = `
<div
  aria-labelledby="headlessui-dialog-title-:r2:"
  aria-modal="true"
  class="relative z-10"
  data-headlessui-state="open"
  data-testid="test-modal"
  id="headlessui-dialog-:r0:"
  role="dialog"
>
  <div
    class="fixed inset-0 bg-black bg-opacity-25 ease-out duration-300 opacity-0"
  />
  <div
    class="fixed inset-0 overflow-y-auto"
  >
    <div
      class="flex min-h-full items-center justify-center p-4 text-center"
    >
      <div
        class="w-full max-w-md transform overflow-hidden rounded-2xl backdrop-blur-xl bg-slate-50 bg-opacity-60 p-6 text-left align-middle shadow-xl transition-all md:text-lg sm:text-md ease-out duration-300 opacity-0 scale-95"
        data-headlessui-state="open"
        id="headlessui-dialog-panel-:r1:"
      >
        <h3
          class="text-center text-2xl font-medium leading-6 mb-4"
          data-headlessui-state="open"
          id="headlessui-dialog-title-:r2:"
        >
          Modal title
        </h3>
        <div>
          modal content
        </div>
        <button>
          Close Modal
        </button>
      </div>
    </div>
  </div>
</div>
`;
